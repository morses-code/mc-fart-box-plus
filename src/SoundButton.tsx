import React from 'react';

interface SoundButtonProps {
  soundFile: string;
}

const SoundButton: React.FC<SoundButtonProps> = ({ soundFile }) => {
  const handleClick = () => {
    const audio = new Audio(soundFile);
    audio.play();
  };

  return (
    <button className="sound-button" data-sound-file={soundFile} onClick={handleClick}>
      ðŸ’©
    </button>
  );
};

export default SoundButton;