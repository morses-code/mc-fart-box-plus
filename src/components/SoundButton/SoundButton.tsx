import React from 'react';

interface SoundButtonProps {
	soundFile: string;
	text: string;
}

const SoundButton: React.FC<SoundButtonProps> = ({ soundFile, text }) => {
	const handleClick = () => {
		const audio = new Audio(soundFile);
		audio.play();
	};

	return (
		<button
			className='sound-button'
			data-sound-file={soundFile}
			onClick={handleClick}
		>
			{text}
		</button>
	);
};

export default SoundButton;
