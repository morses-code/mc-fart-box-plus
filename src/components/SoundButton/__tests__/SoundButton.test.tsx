import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SoundButton from '../SoundButton';

describe('SoundButton', () => {
	const mockPlay = jest.fn();

	beforeEach(() => {
		jest.spyOn(HTMLMediaElement.prototype, 'play').mockImplementation(mockPlay);
	});

	afterEach(() => {
		mockPlay.mockRestore();
	});

	it('plays audio when clicked', () => {
		const soundFile = 'test-sound.mp3';
		const audioPlaySpy = jest.spyOn(window.HTMLMediaElement.prototype, 'play');
		const { getByText } = render(
			<SoundButton
				soundFile={soundFile}
				text='ðŸ’©'
				audio={null}
				setAudio={jest.fn()}
			/>
		);
		const buttonElement = getByText('ðŸ’©');

		fireEvent.click(buttonElement);

		expect(audioPlaySpy).toHaveBeenCalledTimes(1);

		audioPlaySpy.mockRestore();
	});
});
