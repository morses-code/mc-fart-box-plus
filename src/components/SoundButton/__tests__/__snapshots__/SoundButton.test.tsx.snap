// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoundButton plays audio when clicked 1`] = `
<div>
  <button
    class="sound-button"
    data-sound-file="test-sound.mp3"
  >
    ðŸ’©
  </button>
</div>
`;
